sql/core/src/test/resources/sql-tests/inputs/try-string-functions.sql

```diff
@@ -45,4 +45,8 @@ select try_to_binary(null, null);
 select try_to_binary(null, cast(null as string));
 -- invalid format
 select try_to_binary('abc', 1);
-select try_to_binary('abc', 'invalidFormat');
\ No newline at end of file
+select try_to_binary('abc', 'invalidFormat');
+-- format must be foldable
+SELECT try_to_binary(col1, col2) from values ('abc', 'utf-8') as data(col1, col2);
+-- non-foldable input string
+SELECT try_to_binary(col1, 'utf-8') from values ('abc') as data(col1);
```

Comment:
Not strictly relevant to this issue, but I noticed that there were no tests for non-foldable input values, so I added one.